kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionBuilderImpl.java

```diff
@@ -188,25 +189,145 @@ public Transaction build(Engine engine) {
       if (operation == Operation.CREATE_TABLE) {
         throw new TableAlreadyExistsException(table.getPath(engine), "Operation = CREATE_TABLE");
       }
+      return buildTransactionInternal(engine, false /* isNewTableDef */, Optional.of(snapshot));
     } catch (TableNotFoundException tblf) {
       String tablePath = table.getPath(engine);
       logger.info("Table {} doesn't exist yet. Trying to create a new table.", tablePath);
       schema.orElseThrow(() -> requiresSchemaForNewTable(tablePath));
-      // Table doesn't exist yet. Create an initial snapshot with the new schema.
-      Metadata metadata = getInitialMetadata();
-      Protocol protocol = getInitialProtocol();
-      SnapshotQueryContext snapshotContext = SnapshotQueryContext.forVersionSnapshot(tablePath, -1);
-      LogReplay logReplay =
-          getEmptyLogReplay(engine, metadata, protocol, snapshotContext.getSnapshotMetrics());
-      snapshot =
-          new InitialSnapshot(table.getDataPath(), logReplay, metadata, protocol, snapshotContext);
+      return buildTransactionInternal(engine, true /* isNewTableDef */, Optional.empty());
     }
+  }
+
+  /**
+   * Returns a built {@link Transaction} for this transaction builder (with the input provided by
+   * the user) given the provided parameters. This includes validation and updates as defined in the
+   * builder.
+   *
+   * @param isNewTableDef whether we are defining a new table definition or not. This determines
```

Comment:
`whether we are defining a new table definition or not` -> would a simpler variable name be `isCreateOrReplace`?

Basically, what is missing is the exact list of operations that "create a new table definition".  I _think_ it is only create or replace but I am not sure

Reply:
Agree this is cleaner, updating it to be `isCreateOrReplace`. I also cannot think of a different scenario currently and if we come up with one in the future we can always update this internal naming